state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -524,6 +540,23 @@ pub mod pallet {
 	pub type PrewitnessedDepositIdCounter<T: Config<I>, I: 'static = ()> =
 		StorageValue<_, PrewitnessedDepositId, ValueQuery>;
 
+	/// Stores the reporter and the tx_id against the BlockNumber when the report expires.
+	#[pallet::storage]
+	pub(crate) type TaintedTransactions<T: Config<I>, I: 'static = ()> = StorageDoubleMap<
+		_,
+		Identity,
+		T::AccountId,
+		Twox64Concat,
```

Comment:
```suggestion
		Blake2_128Concat,
```

DepositDetails can be freely set by the caller, so we need to be defensive here.